superset/assets/javascripts/components/FilterableTable/FilterableTable.jsx

```diff
@@ -0,0 +1,190 @@
+      for (const k in row) {
+        const val = row[k];
+        if (typeof(val) === 'string') {
+          newRow[k] = val;
+        } else {
+          newRow[k] = JSON.stringify(val);
+        }
+      }
+      return newRow;
+    });
+    return formattedData;
+  }
+
+  getWidthsForColumns() {
+    const PADDING = 40; // accounts for cell padding and width of sorting icon
+    const widthsByColumnKey = {};
+    this.props.orderedColumnKeys.forEach((key) => {
+      const colWidths = this.list.toArray().map(d => getTextWidth(d[key]) + PADDING);
+      // push width of column key to array as well
+      colWidths.push(getTextWidth(key) + PADDING);
+      // set max width as value for key
+      widthsByColumnKey[key] = Math.max(...colWidths);
+    });
+    return widthsByColumnKey;
+  }
+
+  fitTableToWidthIfNeeded() {
+    const containerWidth = this.container.getBoundingClientRect().width;
+    if (containerWidth > this.totalTableWidth) {
+      this.totalTableWidth = containerWidth - 2; // accomodates 1px border on container
+    }
+    this.setState({ fitted: true });
+  }
+
+  componentDidMount() {
+    this.fitTableToWidthIfNeeded();
+  }
+
+  render() {
+    const { sortBy, sortDirection } = this.state;
+    const {
+      filterText,
+      headerHeight,
+      height,
+      orderedColumnKeys,
+      overscanRowCount,
+      rowHeight,
+    } = this.props;
+
+    let sortedAndFilteredList = this.list;
+    // filter list
+    if (filterText) {
+      sortedAndFilteredList = this.list.filter(row => this.hasMatch(filterText, row));
+    }
+    // sort list
+    sortedAndFilteredList = sortedAndFilteredList
+      .sortBy(item => item[sortBy])
+      .update(list => sortDirection === SortDirection.DESC ? list.reverse() : list);
+
+    const rowGetter = ({ index }) => this.getDatum(sortedAndFilteredList, index);
+
+    return (
+      <div
+        style={{ height }}
+        className="filterable-table-container"
+        ref={(ref) => { this.container = ref; }}
+      >
+        {this.state.fitted &&
+          <Table
+            ref="Table"
+            headerHeight={headerHeight}
+            height={height - 2}
+            overscanRowCount={overscanRowCount}
+            rowClassName={this.rowClassName}
+            rowHeight={rowHeight}
+            rowGetter={rowGetter}
+            rowCount={sortedAndFilteredList.size}
+            sort={this.sort}
+            sortBy={sortBy}
+            sortDirection={sortDirection}
+            width={this.totalTableWidth}
+          >
+            {orderedColumnKeys.map((columnKey) => (
+              <Column
+                dataKey={columnKey}
+                disableSort={false}
+                headerRenderer={this.headerRenderer}
+                width={this.widthsForColumnsByKey[columnKey]}
+                label={columnKey}
+                key={columnKey}
+              />
+            ))}
+          </Table>
+        }
+      </div>
+    );
+  }
+
+  getDatum(list, index) {
```

Comment:
I'm used to `render` being the last method at the bottom of the class definition. I think the new linting rules are more picky about method ordering...